lstmcpipe/stages/mc_dl1_to_dl2.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python
+
+# T. Vuillaume,
+# Modifications by E. Garcia
+# lstMCpipe DL1 to DL2 onsite stage (at La Palma cluster)
+
+import os
+import shutil
+from lstmcpipe.io.data_management import (
+    check_and_make_dir_without_verification
+)
+
+
+def dl1_to_dl2(input_dir, path_models, config_file,  particle, wait_jobid_train_pipe=None,
+               wait_jobids_merge=None, dictionary_with_dl1_paths=None, source_environment=None):
+    """
+    Convert onsite files from dl1 to dl2
+
+    Parameters
+    ----------
+    input_dir : str
+        DL1 base path
+    path_models : str
+        Path to the trained models
+    config_file : str
+        Path to a configuration file. If none is given, a standard configuration is applied
+    particle : str
+        Particle to which apply the dl1_to_dl2 stage
+    wait_jobid_train_pipe : str
+        Comma-separated string with the batched jobid from the train stage to indicate the
+        dependencies of the current job to be batched
+    wait_jobids_merge : str
+        string with merge_and_copy jobids
+    dictionary_with_dl1_paths : dict
+        Dictionary with 'particles' as keys containing final output filenames of dl1 files.
+    source_environment : str
+        path to a .bashrc file to source (can be configurable for custom runs @ mc_r0_to_dl3 script) to activate
+        a certain conda environment.
+        ! NOTE : train_pipe AND dl1_to_dl2 MUST BE RUN WITH THE SAME ENVIRONMENT
+
+    Returns
+    -------
+    log_dl1_to_dl2 : dict
+        dictionary of dictionaries containing the jobid of the batched job as key and the run command (the
+        lstchain_mc_dl1_to_dl2 command with all its corresponding arguments) as value.
+
+    jobid_dl1_to_dl2 : str
+        jobid of the batched job to be send (for dependencies purposes) to the next stage of the
+        workflow (dl2_to_irfs)
+
+    """
+
+    output_dir = input_dir.replace('DL1', 'DL2')
+
+    check_and_make_dir_without_verification(output_dir)
+    print(f"\tOutput dir {particle}: {output_dir}")
+
+    log_dl1_to_dl2 = {particle: {}}
+
+    # path to dl1 files by particle type
+    file_list = [dictionary_with_dl1_paths[particle]['training']['train_path_and_outname_dl1'],
+                 dictionary_with_dl1_paths[particle]['testing']['test_path_and_outname_dl1']]
+
+    return_jobids = []
+
+    if wait_jobid_train_pipe == '':
+        wait_jobs = wait_jobids_merge
+    elif wait_jobids_merge == '':
+        wait_jobs = wait_jobid_train_pipe
+    elif wait_jobids_merge == '' and wait_jobid_train_pipe == '':
+        wait_jobs = ''
+    else:
+        wait_jobs = ','.join([wait_jobid_train_pipe, wait_jobids_merge])
+
+    job_name = {'electron': 'dl1-2_e',
+                'gamma': 'dl1-2_g',
+                'gamma-diffuse': 'dl1-2_gd',
+                'proton': 'dl1-2_p',
+                'gamma_off0.0deg': 'g.0_1-2dl',
+                'gamma_off0.4deg': 'g.4_1-2dl'
```

Comment:
Maybe these could be changed to be more consistent, e.g. particle name in front for all cases

Reply:
What I mean is that for offset gammas the particle comes first and for other particles it comes after the dl1-2 part